### DISCUSSION TITLE
GraphQL Java will require Java 11 going forward

### DISCUSSION THREAD

**Discussion by andimarek** [graphql-java/graphql-java#3052] [General]
See https://www.graphql-java.com/blog/java-11-required for the announcement.

Please leave feedback and discuss here.

**Comment by bbakerman** [comment]
We are likely to delay this move until after a non breaking changes 20.1 is released.  After that - Java 11

**Comment by paulbakker** [comment]
We (Netflix) are in the process of moving everything to 17, so this will likely be fine for us. 

I'm curious, though, what's the reason for moving to 11? Is it language features?
If it's mostly to modernize the code base, why not bump to 17, now that other frameworks in the larger ecosystem are making that move as well.

11 seems a bit odd because it is likely going to break a lot of users, but it doesn't *really* move the needle for the language yet.

**Comment by andimarek** [comment]
Hi @paulbakker  ... thanks for your feedback.

I like to think about upgrading to 11 more as an maintenance task. You are right that this is not a major jump in terms of feature or langue in general, but for example even minor things help us: the translation properties can be UTF-8 and we can upgrade the Antlr parser to a new version to name two examples.

In general Java 11 feels like a good middle ground at the moment: GraphQL Java will continue to be conservative with the Java version we develop against, because it is a widely used library, but at the same time we want to be somewhat up to date. 

I hope that make sense.

**Comment by davenice** [comment]
I'm just catching up on this, moving up from a much lower graphql-java in our product.

I was thinking that it seems like quite a short runway to get upgraded - potentially only 6 months between 21 being released and 20 no longer having bug fixes/security remediations?

Is this the timescale you see us ending up with? Particularly given that the upgrade likely requires people changing their infrastructure to accommodate the different JVM it could be tricky. It's not all that straightforward to move our users from Java 8 to Java 11 in service. :-)

**Reply by dondonz** [reply]
Hi @davenice, totally understand that upgrading Java versions can be tricky.

As per our recently announced [release policy](https://www.graphql-java.com/blog/release-policy/), we'll continue to backport security updates for versions for 18 months after release. Hopefully this gives your team enough time to upgrade.

The next release (v21.0) will be early next month (July 2023) and it will be the first version to use Java 11.

**Reply by davenice** [reply]
Thank you - so 20.0 came out in December 2022 which means I think you'd be planning security updates until mid 2024.
